### DISCUSSION TITLE
Scan the same file with multiple managers

### DISCUSSION THREAD

**Discussion by martin-bts** [renovatebot/renovate#12745] [Request Help] [answered]
Renovate works well on my `.gitlab-ci.yml` so far. I want to extend its utility through the regex manager, but apparently renovate will not apply it on top of the existing gitlab manager.

I am using a self-hosted renovate in my Gitlab projects. My main use case is that I use `include` in my `.gitlab-ci.yml` file to reuse yaml files from other projects. For example, the `.gitlab-ci.yml` for my *project mission-one* has:

    include:
      - project: "path/to/some/secret_project"
        ref: "v0.0.7"
        file: "/agents/james.yaml"

and whenever *secret_project* would release a new version, renovate picks up on it and automerges, so that the section now reads

    include:
      - project: "path/to/some/secret_project"
        ref: "v0.0.8"
        file: "/agents/james.yaml"

This is what I want to do and apparently renovate more or less simply does that. In that same `.gitlab-ci.yml` I use variables which contain references to container images, e.g.

    WORKHORSE: "gitlab.my.domain.com/path/to/myimagerepo/image:0.0.7"

I would like for renovate to also pick up on new published images and automatigically change the line above to

    WORKHORSE: "gitlab.my.domain.com/path/to/myimagerepo/image:0.0.8"

as soon as possible. I did some googling and found out that this should be possible using the regex manager, if I add something like this to my renovate.json:

    "regexManagers": [
        {   
          "fileMatch": ["^.gitlab-ci.yml$"],
          "matchStrings": [
            "^\\s+WORKHORSE: ['\\\"]?(?<depName>[^:]*):(?<currentValue>[vV]?\\d+(\\.\\d+)*[^'\\\"]*)['\\\"]?\s*$"
          ],  
          "datasourceTemplate": "docker"
        }   
      ]
Through replacing `\d` with `[[:digit:]]` and removing `?<depName>` and `?<currentValue>` I can test the regex using `grep` and it does find the correct file in `.gitlab-ci.yml`.
Running renovate I can see the following in my logs

> DEBUG: Using file match: ^.gitlab-ci.yml$ for manager regex (repository=path/to/mission-one)
> ...
> DEBUG: Matched 1 file(s) for manager regex: .gitlab-ci.yml (repository=path/to/mission-one)

However, in the dump following

> DEBUG: packageFiles with updates (repository=path/to/mission-one)

my WORKHORSE match does not show up. Why and how do I make this woŕk?

**Comment by rarkins** [comment, marked as answer]
What you're trying is fine (you can match a file with multiple managers) so most likely reason is that the regex extract pattern doesn't work and finds no dependencies to extract.

**Comment by martin-bts** [comment]
I did assume the regex pattern would be applied to the individual line, but is actually applied to the whole file. This is why it works with `grep` but not with `renovate`. On everything else .... oh well ... we will see.

**Reply by viceice** [reply]
We only set `g` modifier for regexs, so `^` and `$` should match line start and end.

https://github.com/renovatebot/renovate/blob/bf093370763892f51bddb59fcb84e39d877564ca/lib/manager/regex/index.ts#L93
https://github.com/renovatebot/renovate/blob/bf093370763892f51bddb59fcb84e39d877564ca/lib/manager/regex/index.ts#L126
https://github.com/renovatebot/renovate/blob/bf093370763892f51bddb59fcb84e39d877564ca/lib/manager/regex/index.ts#L150

**Comment by viceice** [comment]
This is what we use at https://gitlab.com/renovate-bot/renovate-runner/-/blob/main/renovate.json

![image](https://user-images.githubusercontent.com/1798109/142865612-4d976063-90cc-40b3-939b-4d8cab821bd0.png)

**Reply by viceice** [reply]
So i assume you need to change `['\\\"]` to `['\"]`
